src/python/zquantum/core/circuits/_circuit.py

```diff
@@ -39,6 +39,10 @@ def __init__(
         n_qubits: Optional[int] = None,
     ):
         self._operations = list(operations) if operations is not None else []
+
+        if n_qubits and not isinstance(n_qubits, int):
+            raise ValueError("n_qubits must be an integer.")
+
```

Comment:
I think this if won't fire if `n_qubits=0.0`. Since `n_qubits` being 0 does not make sense anyway, WDYT about also testing if `n_qubits` is nonzero?